tests/catch/unit/memory/hipMemcpySync.cc

```diff
@@ -0,0 +1,221 @@
+ * the host.
+ */
+
+using namespace mem_utils;
+
+// value used for memset operations
+constexpr int testValue = 0x11;
+
+
+/*
+ * Set of helper functions handling the different cases for memcpy
+ */
+
+static inline hipMemcpyKind getMemcpyType(allocType type, bool fromHost) {
+  hipMemcpyKind cpyType{};
+  if (fromHost) {
+    switch (type) {
+      case allocType::deviceMalloc:
+        cpyType = hipMemcpyHostToDevice;
+        break;
+      case allocType::devRegistered:
+        cpyType = hipMemcpyHostToDevice;
+        break;
+      default:  // host
+        cpyType = hipMemcpyHostToHost;
+        break;
+    }
+  } else {
+    switch (type) {
+      case allocType::deviceMalloc:
+        cpyType = hipMemcpyDeviceToDevice;
+        break;
+      case allocType::devRegistered:
+        cpyType = hipMemcpyDeviceToDevice;
+        break;
+      default:  // host
+        cpyType = hipMemcpyDeviceToHost;
+        break;
+    }
+  }
+  return cpyType;
+}
+
+template <typename T>
+static inline void memcpyCheck(allocType type, memType memType, T* aPtr, MultiDData& data,
+                               T* fillerData, bool async, hipStream_t stream, bool fromHost) {
+  auto cpyType = getMemcpyType(type, fromHost);
+  auto sizeInBytes = data.pitch * data.getH() * data.getD() * sizeof(T);
+  switch (memType) {
+    case memType::hipMem:
+      if (async) {
+        HIP_CHECK(hipMemcpyAsync(aPtr + data.offset, fillerData, sizeInBytes, cpyType, stream));
+      } else {
+        HIP_CHECK(hipMemcpy(aPtr + data.offset, fillerData, sizeInBytes, cpyType));
+      }
+      break;
+    case memType::hipMem2D:
+      if (async) {
+        HIP_CHECK(hipMemcpy2DAsync(aPtr + data.offset, data.pitch, fillerData, sizeInBytes,
+                                   data.width, data.getH(), cpyType, stream));
+      } else {
+        HIP_CHECK(hipMemcpy2D(aPtr + data.offset, data.pitch, fillerData, sizeInBytes, data.width,
+                              data.getH(), cpyType));
+      }
+      break;
+    case memType::hipMem3D: {
+      hipMemcpy3DParms params{};
+      params.kind = cpyType;
+      params.srcPos = make_hipPos(0, 0, 0);
+      params.dstPos = make_hipPos(data.offset, data.offset, data.offset);
+      params.srcPtr = make_hipPitchedPtr(fillerData, data.width, data.width, data.getH());
+      params.dstPtr = make_hipPitchedPtr(aPtr, data.pitch, data.width, data.getH());
+      hipExtent extent;
+      extent.width = data.width * sizeof(T);
+      extent.height = data.getH();
+      extent.depth = data.getD();
+
+      params.extent = extent;
+      if (async) {
+        HIP_CHECK(hipMemcpy3DAsync(&params, stream));
+      } else {
+        HIP_CHECK(hipMemcpy3D(&params));
+      }
+      break;
+    }
+    default:
+      break;
+  }
+  if (fromHost) {
+    delete[] fillerData;
+  } else {
+    HIP_CHECK(hipFree(fillerData));
+  }
+}
+
+template <typename T> static inline T* createFillerData(size_t count, size_t value, bool fromHost) {
```

Comment:
remove the template from here since the implementation relies on T being char

Reply:
also look at the other template and see if they are also unneeded

Reply:
Looked over and removed unnecessary templates